components/feature/sitepermissions/src/test/java/mozilla/components/feature/sitepermissions/SitePermissionsFeatureTest.kt

```diff
@@ -62,68 +71,110 @@ class SitePermissionsFeatureTest {
+
+        assertFalse(wasCalled)
     }
 
     @Test
-    fun `rejecting a content permission must call reject and consume contentPermissionRequest`() {
+    fun `granting a content permission must call grant and consume contentPermissionRequest`() {
         val session = getSelectedSession()
+        var grantWasCalled = false
 
-        val mockPermissionRequest: PermissionRequest = mock()
+        val permissionRequest: PermissionRequest = object : PermissionRequest {
+            override val uri: String?
+                get() = "http://www.mozilla.org"
+            override val permissions: List<Permission>
+                get() = listOf(ContentGeoLocation())
 
-        sitePermissionFeature.start()
+            override fun grant(permissions: List<Permission>) {
+                grantWasCalled = true
+            }
 
-        session.contentPermissionRequest = Consumable.from(mockPermissionRequest)
+            override fun reject() = Unit
+        }
 
-        sitePermissionFeature.onContentPermissionDeny(session.id, session.url)
+        session.contentPermissionRequest = Consumable.from(permissionRequest)
 
-        verify(mockPermissionRequest).reject()
+        val prompt = sitePermissionFeature.onContentPermissionRequested(session, permissionRequest)
+
+        val positiveButton = prompt.buttons.find { it.positive }
+        positiveButton?.onClick?.invoke()
+
+        assertTrue(grantWasCalled)
         assertTrue(session.contentPermissionRequest.isConsumed())
     }
 
     @Test
-    fun `rejecting a content permission with an unknown session will no consume pending contentPermissionRequest`() {
+    fun `rejecting a location content permission must call reject and consume contentPermissionRequest`() {
         val session = getSelectedSession()
-        val mockPermissionRequest: PermissionRequest = mock()
+        var grantWasCalled = false
 
-        sitePermissionFeature.start()
+        val permissionRequest: PermissionRequest = object : PermissionRequest {
+            override val uri: String?
+                get() = "http://www.mozilla.org"
+            override val permissions: List<Permission>
+                get() = listOf(ContentGeoLocation())
 
-        session.contentPermissionRequest = Consumable.from(mockPermissionRequest)
+            override fun reject() {
+                grantWasCalled = true
+            }
+
+            override fun grant(permissions: List<Permission>) = Unit
+        }
 
-        assertFalse(session.contentPermissionRequest.isConsumed())
+        session.contentPermissionRequest = Consumable.from(permissionRequest)
 
-        sitePermissionFeature.onContentPermissionDeny("unknown_session", session.url)
+        val prompt = sitePermissionFeature.onContentPermissionRequested(session, permissionRequest)
 
-        assertFalse(session.contentPermissionRequest.isConsumed())
+        val negativeButton = prompt.buttons.find { !it.positive }
+        negativeButton?.onClick?.invoke()
```

Comment:
nit: In tests when you expect something to exist then let's test explicitly: `negativeButton!!.onClick!!.invoke()`. If something in this chain does not exist it's good to fail early.

Reply:
👍